code/data_setup/parse_imagenet_bb.py

```diff
@@ -3,19 +3,100 @@
 import sys
 import os
 import itertools
+import pandas as pd
 import xml.etree.ElementTree as ET 
 
-def parse_file(bbox_dir, filepath): 
-    tree = ET.parse(bbox_dir + filepath)
-    print root.findall('./')
-    pass
+def parse_file(bbox_dir, filename): 
+    """Parse an individual XML file and grab the relevant elements. 
 
+    Args: 
+    ----
+        bbox_dir: str
+            Holds the directory path to the XML file. 
+        filename: str
+            Holds the name of the XML file. To be attached to the 
+            `bbox_dir` to be read in and parsed. 
+
+    Output: 
+    ------
+        output_lst: list of lists 
+            Holds lists of the values parsed from an XML file. 
+    """
+    tree = ET.parse(bbox_dir + filename)
+    root = tree.getroot()
+
+    output_lst = []
+    width, height = parse_size(root)
+
+    for child in root.findall('./'): 
+        if child.tag == "object": 
+
+            name = child.find('name').text
+            subcategory = child.find('subcategory')
+            subcategory = None if subcategory is None else subcategory.text
+
+            bounding_box = child.find('bndbox')
+            xmin, xmax, ymin, ymax = parse_bb(bounding_box)
+
+            object_lst = [width, height, name, subcategory, xmin, 
+                    xmax, ymin, ymax]
+
+            output_lst.append(object_lst)
+
+    return output_lst
+
+def parse_size(root): 
+    """Parse the width and height of the image out of the root node. 
+
+    Args: 
+    ----
+        root: xml.etree.ElementTree.Element
+
+    Output: 
+    ------
+        width: str
+        height str
+    """
+
+    size_node = root.find('size')
+    width = size_node.find('width').text
+    height = size_node.find('height').text
+
+    return width, height
+
+def parse_bb(bounding_box): 
+    """Parse the coordinates of the bounding box from the bounding box node.
+
+    Args: 
+    ----
+        bounding_box: xml.etree.ElementTree.Element
+
+    Output: 
+    ------
+        xmin: str
+        xmax: str
+        ymin: str
+        ymax: str
+    """
+    
+    xmin = bounding_box.find('xmin').text
+    xmax = bounding_box.find('xmax').text
+    ymin = bounding_box.find('ymin').text
+    ymax = bounding_box.find('ymax').text
+
+    return xmin, xmax, ymin, ymax
+    
 if __name__ == '__main__': 
     bbox_dir = sys.argv[1]
+    output_filepath = sys.argv[2]
     xml_files_by_dir = (i[2] for i in os.walk(bbox_dir))
     bbox_xml_filenames = itertools.chain(*xml_files_by_dir)
 
-    parsed_results = [parse_file(bbox_dir, filename) for \
-            filename in bbox_xml_filenames]
-
+    all_bboxes = (parse_file(bbox_dir, filename) for filename in bbox_xml_filenames)
+    end_lst_iterator = itertools.chain(*all_bboxes)
+    end_lst = [bbox_lst for bbox_lst in end_lst_iterator]
```

Comment:
Is this just converting the iterator to a list?

If so, you can do that with `end_lst = list(end_lst_iterator)` 
I think that makes it more obvious what that line is doing.  Or for that matter you could just wrap the `itertools.chain` line above it in a call to `list()`

Either way, I think this is good to merge.